Add docs for installing nginx-agent from repository (#309)
nginx/agent: hugo/content/installation-plus.md

Reviewer: Need to add FreeBSD instructions https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/#installing-nginx-plus-on-freebsd

---
title: "Installation From NGINX Plus Repository"
draft: false
description: "Learn how to install NGINX Agent from NGINX Plus repository."
weight: 400
toc: true
tags: [ "docs" ]
docs: "DOCS-1092"
categories: ["configuration"]
doctypes: ["task"]
---

## Install NGINX Plus

NGINX Agent interfaces directly with an NGINX server process installed on the same system. If you don't have it already, follow these steps to install [NGINX Plus](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/). Once installed, ensure the NGINX instance is running.

## Install NGINX Agent from NGINX Plus Repository

> First ensure that you are running a supported distribution and architecture: [technical specifications](../technical-specifications/)

Reviewer 2: ```suggestion
First, ensure that you are running a supported distribution and architecture: [technical specifications](../technical-specifications/)
```

    Acquire::https::pkgs.nginx.com::SslKey      "/etc/ssl/nginx/nginx-repo.key";
    ```

  - To install `nginx-agent`, run the following commands:
    ```
    sudo apt update
    sudo apt install nginx-agent
    ```

### SLES
  - Create the `/etc/ssl/nginx` directory:
    ```
    sudo mkdir -p /etc/ssl/nginx
    ```

  - Log in to [MyF5 Customer Portal](https://account.f5.com/myf5/) and download your `nginx-repo.crt` and `nginx-repo.key` files.

  - Copy the files to the `/etc/ssl/nginx/` directory:
    ```
    sudo cp nginx-repo.crt nginx-repo.key /etc/ssl/nginx/
    ```

  - Create a file bundle of the certificate and key:
    ```
    cat /etc/ssl/nginx/nginx-repo.crt /etc/ssl/nginx/nginx-repo.key > /etc/ssl/nginx/nginx-repo-bundle.crt
    ```

  - Install the prerequisites:
    ```
    sudo zypper install curl ca-certificates gpg2 gawk
    ```

  - To set up the zypper repository for `nginx-agent` packages, run the following command:
    ```
    sudo zypper addrepo --refresh --check \
        'https://pkgs.nginx.com/nginx-agent/sles/$releasever_major?ssl_clientcert=/etc/ssl/nginx/nginx-repo-bundle.crt&ssl_verify=peer' nginx-agent
    ```

>   - Next, import an official nginx signing key so `zypper`/`rpm` could verify the packages authenticity. Fetch the key:

Reviewer 2: ```suggestion
  - Next, import an official NGINX signing key so `zypper`/`rpm` can verify the package's authenticity. Fetch the key:
```

          Key fingerprint = 573B FD6B 3D8F BC64 1079  A6AB ABF5 BD82 7BD9 BF62
    uid nginx signing key <signing-key@nginx.com>
    ```

  - Finally, import the key to the rpm database:
    ```
    sudo rpmkeys --import /tmp/nginx_signing.key

  - To install `nginx-agent`, run the following command:
    ```
    sudo zypper install nginx-agent
    ```

### Alpine Linux
  - Log in to [MyF5 Customer Portal](https://account.f5.com/myf5/) and download your `nginx-repo.crt` and `nginx-repo.key` files.

  - Copy the files to the `/etc/apk/` directory:
    ```
    sudo cp nginx-repo.crt nginx-repo.key /etc/apk/
    ```
    <!-- ```
    sudo cp nginx-repo.key /etc/apk/cert.key
    sudo cp nginx-repo.crt /etc/apk/cert.pem
    ``` -->

  - Install the prerequisites:
    ```
    sudo apk add openssl curl ca-certificates
    ```

  - To set up the apk repository for `nginx-agent` packages, run the following command:
    ```
    printf "%s%s%s%s\n" \
        "@nginx-agent " \
        "https://pkgs.nginx.com/nginx-agent/alpine/v" \
        `grep -o -E '^[0-9]+\.[0-9]+' /etc/alpine-release` \
        "/main" \
        | sudo tee -a /etc/apk/repositories
    ```

>   - Next, import an official nginx signing key so apk could verify the packages authenticity. Fetch the key:

Reviewer 2: ```suggestion
  - Next, import an official NGINX signing key so apk can verify the package's authenticity. Fetch the key:
```

- [Installing NGINX Agent on Ubuntu](#installing-nginx-agent-on-ubuntu)
- [Installing NGINX Agent on Debian](#installing-nginx-agent-on-debian)
- [Installing NGINX Agent on SLES](#installing-nginx-agent-on-sles)
- [Installing NGINX Agent on Alpine Linux](#installing-nginx-agent-on-alpine-linux)
- [Installing NGINX Agent on Amazon Linux](#installing-nginx-agent-on-amazon-linux)

### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux

1. Create the `/etc/ssl/nginx` directory:

    ```shell
    sudo mkdir -p /etc/ssl/nginx
    ```

1. Log in to [MyF5 Customer Portal](https://account.f5.com/myf5/) and download your `nginx-repo.crt` and `nginx-repo.key` files.

1. Copy the files to the `/etc/ssl/nginx/` directory:

    ```shell
    sudo cp nginx-repo.crt nginx-repo.key /etc/ssl/nginx/
    ```

1. Install the prerequisites:

    ```shell
    sudo yum install yum-utils procps
    ```

1. To set up the yum repository, create the file named `/etc/yum.repos.d/nginx-agent.repo` with the following contents:

> 

Reviewer: Need to add command to this step create repo file in that directory. See step 2 from existing doc https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/#installing-a-prebuilt-rhelcentosoracle-linuxalmalinuxrocky-linux-package-from-the-official-nginx-repository
``` shell
 sudo vi /etc/yum.repos.d/nginx-agent.repo

```

---

> title: "Install NGINX Agent From NGINX Plus Repository"

Reviewer: Title to change to "Installation from NGINX Plus Repository" to be in line with the OSS title.

draft: false
description: "Learn how to install NGINX Agent from NGINX Plus repository."
weight: 400
toc: true
tags: [ "docs" ]
docs: "DOCS-1217"
categories: ["configuration"]
doctypes: ["task"]
---

## Prerequisites

- An NGINX Plus subscription (purchased or trial)
- NGINX Plus installed. Once installed, ensure it is running. If you don't have it installed already, follow these steps to install [NGINX Plus](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-plus/)
- A [supported operating system and architecture](../technical-specifications/#supported-distributions)
- `root` privilege
- Your credentials to the MyF5 Customer Portal, provided by email from F5, Inc.
- Your NGINX Plus certificate and public key (`nginx-repo.crt` and `nginx-repo.key` files), provided by email from F5, Inc.

## Configure NGINX Plus Repository for installing NGINX Agent

Before you install NGINX Agent for the first time on your system, you need to set up the `nginx-agent` packages repository. Afterward, you can install and update NGINX Agent from the repository.

> - [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)

Reviewer 2: ```suggestion
- [Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux](#installing-nginx-agent-on-rhel-centos-rocky-linux-almalinux-and-oracle-linux)
```

- [Installing NGINX Agent on Ubuntu](#installing-nginx-agent-on-ubuntu)
- [Installing NGINX Agent on Debian](#installing-nginx-agent-on-debian)
- [Installing NGINX Agent on SLES](#installing-nginx-agent-on-sles)
- [Installing NGINX Agent on Alpine Linux](#installing-nginx-agent-on-alpine-linux)
- [Installing NGINX Agent on Amazon Linux](#installing-nginx-agent-on-amazon-linux)
- [Installing NGINX Agent on FreeBSD](#installing-nginx-agent-on-freebsd)

> ### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux and Oracle Linux

Reviewer 2: ```suggestion
### Installing NGINX Agent on RHEL, CentOS, Rocky Linux, AlmaLinux, and Oracle Linux
```
